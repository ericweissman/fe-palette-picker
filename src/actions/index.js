export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

//projects
export const getProjectsSuccess = (projects) => ({
  type: 'GET_PROJECTS_SUCCESS',
  projects
})

export const addProjectSuccess = (project) => ({
  type: 'ADD_PROJECT_SUCCESS',
  project
})