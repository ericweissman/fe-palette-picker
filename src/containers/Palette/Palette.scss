.small-color {
  height: 100px;
  width: 100px;
}

.palette-small {
  display: flex;
}