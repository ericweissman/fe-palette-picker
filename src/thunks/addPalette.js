import { isLoading, hasErrored, addPaletteSuccess } from '../actions'

export const addPalette = (url, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addPaletteSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}