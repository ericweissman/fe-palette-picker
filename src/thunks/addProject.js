import { isLoading, hasErrored, addProjectSuccess } from '../actions'

export const addProject = (url, project_name) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type" : "application/json"
        },
        body: JSON.stringify({ project_name })
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addProjectSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}