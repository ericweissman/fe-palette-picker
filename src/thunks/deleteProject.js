import { isLoading, hasErrored, deleteProjectSuccess } from '../actions'

export const deleteProject = (url, projectID) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "DELETE",
        headers: {
          "Content-Type" : "application/json"
        }
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      // const result = await response.json()
      dispatch(deleteProjectSuccess(projectID))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}