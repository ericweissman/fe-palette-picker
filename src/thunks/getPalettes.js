import { isLoading, hasErrored, getPalettesSuccess } from '../actions'

export const getPalettes = (url, id) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(getPalettesSuccess(result, id))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}