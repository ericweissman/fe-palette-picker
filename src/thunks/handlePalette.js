import { isLoading, hasErrored } from '../actions'

export const handlePalette = (url, actionToDispatch, method, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: `${method}`,
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(actionToDispatch(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}