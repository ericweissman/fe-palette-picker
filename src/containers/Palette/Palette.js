import React, { Component } from 'react';
import { connect } from 'react-redux'
import { editPaletteSuccess } from '../../actions';
import { handlePalette } from '../../thunks/handlePalette';
import PropTypes from 'prop-types'

export class Palette extends Component {
  state = {
    edited: false,
    name: '',
  }

  editColors = () => {
    const { palette } = this.props
    const colors = [palette.color_1, palette.color_2, palette.color_3, palette.color_4, palette.color_5]
    this.props.setActive(colors)
    this.props.editPalette({ palette_name: palette.palette_name, id: palette.id, project_id: palette.project_id })
  }

  toggleEdited = () => {
    const { name } = this.state
    const { id, project_id, color_1, color_2, color_3, color_4, color_5 } = this.props.palette
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${project_id}/palettes/${id}`
    const palette = {
      palette_name: name,
      color_1,
      color_2,
      color_3,
      color_4,
      color_5,
    }
    this.setState({ edited: !this.state.edited })
    if (this.state.edited) {
      this.props.handlePalette(url, editPaletteSuccess, 'PUT', palette)
    }
  }

  updateName = (e) => {
    const { value } = e.target
    this.setState({
      name: value
    })
  }


  render() {
    const { deletePalette, palette, setActive } = this.props
    const colors = [palette.color_1, palette.color_2, palette.color_3, palette.color_4, palette.color_5]
    const { edited } = this.state
    return (
      <div className='palette'>
        {
          edited ?
            <input onChange={this.updateName} value={this.state.name}></input> :
            <h5 className='palette-name' onClick={() => setActive(colors)}>{palette.palette_name}</h5>
        }
        <button onClick={this.toggleEdited}>{edited ? 'Save' : 'Edit'}</button>
        <div className='palette-small'>
          {
            colors.map((color, i) => {
              return (
                <div key={i} onClick={() => setActive(colors)} className='small-color' style={{ backgroundColor: color }}> {color} </div>
              )
            })
          }
        </div >
        <button onClick={() => deletePalette(palette.id)}>X</button>
        <button onClick={this.editColors}> Edit</button>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handlePalette: (url, actionToDispatch, method, palette) => dispatch(handlePalette(url, actionToDispatch, method, palette)),
})

export default connect(null, mapDispatchToProps)(Palette);

Palette.propTypes = {
  palette: PropTypes.object,
  setActive: PropTypes.func,
  deletePalette: PropTypes.func,
  editPalette: PropTypes.func,
  editName: PropTypes.func,
}