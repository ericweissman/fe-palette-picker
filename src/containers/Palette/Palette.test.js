import React from 'react'
import { shallow } from 'enzyme'
import { Palette, mapDispatchToProps } from './Palette'
import { editPaletteSuccess } from '../../actions';
import { handlePalette } from '../../thunks/handlePalette';

jest.mock('../../thunks/handlePalette')

describe('Palette', () => {
  let wrapper
  const mockPalette = {
    palette_name: 'Cool Palette',
    color_1: 'AAAAA',
    color_2: 'AAAAA',
    color_3: 'AAAAA',
    color_4: 'AAAAA',
    color_5: 'AAAAA'
  }
  const mockProps = {
    palette: mockPalette,
    setActive: jest.fn(),
    deletePalette: jest.fn(),
    editPalette: jest.fn(),
    editName: jest.fn(),
    handlePalette: jest.fn(),
  }

  it('should match the correct snapshot when edited is false', () => {
    wrapper = shallow(<Palette {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the correct snapshot when edited is true', () => {
    wrapper.setState({ edited: true })
    wrapper = shallow(<Palette {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the proper default state', () => {
    const expectedState = {
      edited: false,
      name: ''
    }
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should call setActive and editPalette when editColors is called', () => {
    wrapper.instance().editColors()
    expect(mockProps.setActive).toHaveBeenCalled()
    expect(mockProps.editPalette).toHaveBeenCalled()
  })

  it('should update the palette name in state', () => {
    const mockEvent = { target: { name: 'name', value: 'New Palette' } }
    const expectedState = { edited: false, name: 'New Palette' }
    wrapper.instance().updateName(mockEvent)
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should toggle the edited state and call handlePalette', () => {
    const expectedState = { edited: true, name: 'New Palette' }
    wrapper.instance().toggleEdited()
    expect(wrapper.state()).toEqual(expectedState)
    expect(mockProps.handlePalette).toHaveBeenCalled()
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with handlePalette', () => {
      const mockDispatch = jest.fn()
      const mockURL = 'http//localhost:1234'
      const mockPalette = {
        palette_name: 'Cool Palette',
        color_1: 'AAAAA',
        color_2: 'AAAAA',
        color_3: 'AAAAA',
        color_4: 'AAAAA',
        color_5: 'AAAAA'
      }
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = handlePalette(mockURL, editPaletteSuccess, 'PUT', mockPalette)

      mappedProps.handlePalette(mockURL, editPaletteSuccess, 'PUT', mockPalette)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})