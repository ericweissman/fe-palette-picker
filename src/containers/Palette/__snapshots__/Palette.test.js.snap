// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the correct snapshot when edited is false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Palette
    deletePalette={[MockFunction]}
    editName={[MockFunction]}
    editPalette={[MockFunction]}
    handlePalette={[MockFunction]}
    palette={
      Object {
        "color_1": "AAAAA",
        "color_2": "AAAAA",
        "color_3": "AAAAA",
        "color_4": "AAAAA",
        "color_5": "AAAAA",
        "palette_name": "Cool Palette",
      }
    }
    setActive={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="palette-name-area"
        >
          <h5
            className="palette-name"
            onClick={[Function]}
          >
            Cool Palette
          </h5>
          <button
            className="edit-palette-name-btn"
            onClick={[Function]}
          />
        </div>,
        <div
          className="palette-small"
        >
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <button
            className="edit-palette-btn"
            onClick={[Function]}
          />
          <button
            className="delete-palette-btn"
            onClick={[Function]}
          />
        </div>,
      ],
      "className": "palette",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="palette-name"
              onClick={[Function]}
            >
              Cool Palette
            </h5>,
            <button
              className="edit-palette-name-btn"
              onClick={[Function]}
            />,
          ],
          "className": "palette-name-area",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cool Palette",
              "className": "palette-name",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cool Palette",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "edit-palette-name-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
            ],
            <button
              className="edit-palette-btn"
              onClick={[Function]}
            />,
            <button
              className="delete-palette-btn"
              onClick={[Function]}
            />,
          ],
          "className": "palette-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "edit-palette-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "delete-palette-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="palette-name-area"
          >
            <h5
              className="palette-name"
              onClick={[Function]}
            >
              Cool Palette
            </h5>
            <button
              className="edit-palette-name-btn"
              onClick={[Function]}
            />
          </div>,
          <div
            className="palette-small"
          >
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <button
              className="edit-palette-btn"
              onClick={[Function]}
            />
            <button
              className="delete-palette-btn"
              onClick={[Function]}
            />
          </div>,
        ],
        "className": "palette",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="palette-name"
                onClick={[Function]}
              >
                Cool Palette
              </h5>,
              <button
                className="edit-palette-name-btn"
                onClick={[Function]}
              />,
            ],
            "className": "palette-name-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cool Palette",
                "className": "palette-name",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cool Palette",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "edit-palette-name-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
              ],
              <button
                className="edit-palette-btn"
                onClick={[Function]}
              />,
              <button
                className="delete-palette-btn"
                onClick={[Function]}
              />,
            ],
            "className": "palette-small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "edit-palette-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "delete-palette-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Palette should match the correct snapshot when edited is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Palette
    deletePalette={[MockFunction]}
    editName={[MockFunction]}
    editPalette={[MockFunction]}
    handlePalette={[MockFunction]}
    palette={
      Object {
        "color_1": "AAAAA",
        "color_2": "AAAAA",
        "color_3": "AAAAA",
        "color_4": "AAAAA",
        "color_5": "AAAAA",
        "palette_name": "Cool Palette",
      }
    }
    setActive={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="palette-name-area"
        >
          <h5
            className="palette-name"
            onClick={[Function]}
          >
            Cool Palette
          </h5>
          <button
            className="edit-palette-name-btn"
            onClick={[Function]}
          />
        </div>,
        <div
          className="palette-small"
        >
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <div
            className="small-color"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "AAAAA",
              }
            }
          />
          <button
            className="edit-palette-btn"
            onClick={[Function]}
          />
          <button
            className="delete-palette-btn"
            onClick={[Function]}
          />
        </div>,
      ],
      "className": "palette",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="palette-name"
              onClick={[Function]}
            >
              Cool Palette
            </h5>,
            <button
              className="edit-palette-name-btn"
              onClick={[Function]}
            />,
          ],
          "className": "palette-name-area",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cool Palette",
              "className": "palette-name",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cool Palette",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "edit-palette-name-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
              <div
                className="small-color"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "AAAAA",
                  }
                }
              />,
            ],
            <button
              className="edit-palette-btn"
              onClick={[Function]}
            />,
            <button
              className="delete-palette-btn"
              onClick={[Function]}
            />,
          ],
          "className": "palette-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "className": "small-color",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "AAAAA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "edit-palette-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "delete-palette-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="palette-name-area"
          >
            <h5
              className="palette-name"
              onClick={[Function]}
            >
              Cool Palette
            </h5>
            <button
              className="edit-palette-name-btn"
              onClick={[Function]}
            />
          </div>,
          <div
            className="palette-small"
          >
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <div
              className="small-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "AAAAA",
                }
              }
            />
            <button
              className="edit-palette-btn"
              onClick={[Function]}
            />
            <button
              className="delete-palette-btn"
              onClick={[Function]}
            />
          </div>,
        ],
        "className": "palette",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="palette-name"
                onClick={[Function]}
              >
                Cool Palette
              </h5>,
              <button
                className="edit-palette-name-btn"
                onClick={[Function]}
              />,
            ],
            "className": "palette-name-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cool Palette",
                "className": "palette-name",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cool Palette",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "edit-palette-name-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
                <div
                  className="small-color"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "AAAAA",
                    }
                  }
                />,
              ],
              <button
                className="edit-palette-btn"
                onClick={[Function]}
              />,
              <button
                className="delete-palette-btn"
                onClick={[Function]}
              />,
            ],
            "className": "palette-small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "className": "small-color",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "AAAAA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "edit-palette-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "delete-palette-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
